http_outlet.go

```diff
@@ -136,13 +136,13 @@ func (h *HTTPOutlet) retryPost(batch Batch) {
 	}
 }
 
-func (h *HTTPOutlet) post(formatter HTTPFormatter, uuid string) error {
+func (h *HTTPOutlet) post(formatter HTTPFormatter) error {
 	req, err := formatter.Request()
 	if err != nil {
 		return err
 	}
 
-	req.Header.Add("X-Request-Id", uuid)
+	uuid := req.Header.Get("X-Request-Id")
```

Comment:
The only reason we extract it here is that it's used in some logging below. It's part of the request, which the formatter generates from the batch, so this seems like the right place / way to get it.